lib/gears/surface.lua

```diff
@@ -205,6 +206,55 @@ function surface.apply_shape_bounding(draw, shape, ...)
   draw.shape_bounding = img._native
 end
 
+local function run_in_hierarchy(self, cr, width, height)
+    local function main_layout(h)
+        h:update({dpi=96}, self, width, height)
```

Comment:
Hm... apparently I misunderstood your intention with this function.
In the following code, I'd espect a SVG saying "foo" and not "bar":
```
local t = wibox.widget.textbox("foo")
gears.surface.widget_to_svg(t, "/tmp/t.svg", 100, 100)
t:set_text("bar")
```
Apparently, you want it to say `bar`. And mentioned something like "multiple frames" being produced. I don't really get this.